lib/models/room.js

```diff
@@ -401,6 +410,27 @@ Room.prototype.addReceipt = function(event) {
     });
 };
 
+/**
+ * Update the room-tag event for the room.  The previous one is overwritten.
+ * @param {MatrixEvent} event the m.tag event
+ */
+Room.prototype.addTags = function(event) {
+    // event content looks like:
+    // content: {
+    //    tags: {
+    //       $tagname: { $metadata: $value },
+    //       $tagname: { $metadata: $value },
+    //    }
+    // }
+
+    // XXX: we could do a deep-comparison to see if the tags have really
+    // changed - but do we want to bother?
+    this.emit("Room.tags", this);
```

Comment:
I agree it sounds like a lot of work without any concrete gains. Devs can always do this manually if they care (though the reverse (only emit if deeply changed) would be impossible to detect this case since they need to know that a room has got tags in order to update!)